controller/configentry_controller.go

```diff
@@ -31,13 +30,13 @@ const (
 // ConfigEntryController to abstract CRD-specific controllers.
 type Controller interface {
 	// Update updates the state of the whole object.
-	Update(context.Context, runtime.Object, ...client.UpdateOption) error
+	Update(context.Context, client.Object, ...client.UpdateOption) error
 	// UpdateStatus updates the state of just the object's status.
-	UpdateStatus(context.Context, runtime.Object, ...client.UpdateOption) error
+	UpdateStatus(context.Context, client.Object, ...client.UpdateOption) error
 	// Get retrieves an obj for the given object key from the Kubernetes Cluster.
 	// obj must be a struct pointer so that obj can be updated with the response
 	// returned by the Server.
-	Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object) error
```

Comment:
This is because the interface of `Reconciler` and the controller have been updated to require `client.Object` and not `runtime.Object` which required the `ConfigEntryResource` interface to need an update